weld/llvm.rs

```diff
@@ -765,223 +781,279 @@ impl LlvmGenerator {
         Ok(())
     }
 
+    /*********************************************************************************************
+    //
+    //
+    //
+    // TODO(shoumik): Refactor code above this point!
+    //
+    //
+    //
+    ********************************************************************************************/
+
+    /*********************************************************************************************
+    //
+    // Utilities
+    //
+    *********************************************************************************************/
+
+    /// Returns the Type, LLVM type (as a string), and LLVM symbol name (as a string).
+    fn symbol_info<'a>(&mut self, func: &'a SirFunction, sym: &Symbol) -> WeldResult<(&'a Type, String, String)> {
+        let ty = func.symbol_type(sym)?;
+        let llvm_ty = self.llvm_type(ty)?;
+        let llvm_str = llvm_symbol(sym);
+        Ok((ty, llvm_ty, llvm_str))
+    }
```

Comment:
Okay, I changed this to `llvm_type_and_name`, and then you call `symbol_type` explicitly if you need the type. It's a bit weird since `llvm_type_and_name` needs to call `symbol_type` too, but it's more readable.